Sources/MongoMobile/MongoMobile.swift

```diff
@@ -75,25 +75,63 @@ private func getStatusExplanation(_ status: OpaquePointer?) -> String {
 
 /// A class containing static methods for working with MongoMobile.
 public class MongoMobile {
-    private static var libraryInstance: OpaquePointer?
+    public static let shared = MongoMobile()
+
+    private var libraryInstance: OpaquePointer?
     /// Cache embedded instances for cleanup
-    private static var embeddedInstances = [String: OpaquePointer]()
+    private var embeddedInstances = [String: OpaquePointer]()
     /// Cache embedded clients for cleanup
-    private static var embeddedClients = [WeakRef<MongoClient>]()
+    private var embeddedClients = [WeakRef<MongoClient>]()
+
+    private init() {
+        #if os(Linux)
+        let observer = CFRunLoopObserverCreateWithHandler(
+            nil, CFRunLoopActivity.exit.rawValue, true, 0, { (observer, activity) in
+            try? MongoMobile.shared.destroy()
+            MongoSwift.cleanup()
+        })
+        CFRunLoopAddObserver(CFRunLoopGetMain(), observer, CFRunLoopMode.defaultMode)
+        #else
+        #if swift(>=4.2)
+        // observe when application will terminate
+        NotificationCenter.default.addObserver(
+            self,
+            selector: #selector(cleanup),
+            name: UIApplication.willTerminateNotification,
+            object: nil
+        )
+        #else
+        // observe when application will terminate
+        NotificationCenter.default.addObserver(
+            self,
+            selector: #selector(cleanup),
+            name: NSNotification.Name.UIApplicationWillTerminate,
+            object: nil
+        )
+        #endif
+        #endif
+        MongoSwift.initialize()
+        reinitialize()
+    }
+
+    #if !os(Linux)
+    @objc internal func cleanup() {
+        try? MongoMobile.shared.destroy()
+        MongoSwift.cleanup()
+    }
+    #endif
 
     /**
      * Perform required operations to initialize the embedded server.
      */
-    public static func initialize() throws {
-        MongoSwift.initialize()
-
+    public func reinitialize() {
```

Comment:
can we update the docstring to better reflect when users need to call this? "reinitializes state after `destroy` has been called" or something